### ISSUE TITLE
Adobe Acrobat is not able to open the PDF and gives me error code 110

### ISSUE THREAD

**Issue by sumit-022** [diegomura/react-pdf#2686]
<img width="957" alt="image" src="https://github.com/diegomura/react-pdf/assets/96455464/cb521e1a-4dd5-4668-a7ae-4cbaf6b1ba4f">

Chrome or any other browser is able to open the pdf but Adobe Acrobat can't

<img width="480" alt="image" src="https://github.com/diegomura/react-pdf/assets/96455464/7c692a9c-3905-4b6f-9d80-de5aeb51d67b">

This was the error on verifying the pdf generated, online

**Comment by joelybahh** [comment]
Could be related to jpeg compatibility due to `this.bits` being undefined, which should be fixed via either of these PRs. It would need to be re-generated with this fix/update deployed.

- [https://github.com/diegomura/react-pdf/pull/2646](https://github.com/diegomura/react-pdf/pull/2646)
- [https://github.com/diegomura/react-pdf/pull/2689](https://github.com/diegomura/react-pdf/pull/2689)

**Comment by joelybahh** [comment]
Until these are fixed, some workarounds do exist that have varying levels of applicability depending on your context.

If you're in a situation to convert the images ahead of time to PNG, it _could_ fix it (Assuming this does in fact relate to the jpg issue).

If you're not, you could also use a runtime image conversion tool like `sharp`, but that requires server side PDF generation so also isn't applicable to everyone.

**Comment by knuula** [comment]
@diegomura is there anything that can be done to get these last 2 checks to pass?

I am using Node 18 and React 17 at the moment.


![image](https://github.com/diegomura/react-pdf/assets/48263034/1607d0f5-9816-4b46-8654-12d1fe6a15d1)

**Comment by knuula** [comment]
I went ahead and updated to React 18. I am trying to update to the latest @react-pdf/render. The version that gets loaded in does not have the changes shown in #2689 

Am I missing something here? How to I get that working? @joelybahh

**Comment by joelybahh** [comment]
@knuula I can't from latest version either, unless this PR is yet to deploy changes to NPM? I compared my local dist to the one in node_modules post upgrade and they're not the same so to me it seems like the changes on yesterdays push didn't include my fix yet

**Comment by knuula** [comment]
@joelybahh okay good to know I'm not the only one on this. Thought I was going crazy.

I was just looking at npm and it looks like the version on there was last changed on 3/25/2024 at 7:26pm (CST). So it is definitely not up to date with the latest version that is here on GitHub.

any way @diegomura can push this through to npm?

**Comment by joelybahh** [comment]
I confirmed my fix works by just copying my local `lib` folder into `node_modules/@react-pdf/pdfkit/lib` of my other project.

You'll know if you have the jpeg fix if you search inside `node_modules/@react-pdf/pdfkit/lib/pdfkit.js` for `if (marker.name === 'SOF')`, and the lines under that look like the below:

```javascript
if (marker.name === 'SOF') {
        this.bits || (this.bits = marker.precision);
        this.width || (this.width = marker.width);
        this.height || (this.height = marker.height);
        this.colorSpace || (this.colorSpace = COLOR_SPACE_MAP[marker.numberOfComponents]);
      }
```

My only guess is it hasn't deployed those changes yet, or didn't build

**Comment by joelybahh** [comment]
@knuula I think the release yesterday was an automated package version bump, as my PR was merged 12 hours ago, but the release yesterdat appears to have been before that, my guess is this change is not on NPM yet.

**Comment by coren-frankel** [comment]
4.3.2 was released on npm ~10 hours ago and my PDF is coming out just as expected now.

> Could be related to jpeg compatibility due to `this.bits` being undefined, which should be fixed via either of these PRs. It would need to be re-generated with this fix/update deployed.
> 
> * [Fix invalid JPEG image #2646](https://github.com/diegomura/react-pdf/pull/2646)
> * [fix: jpeg precision metadata #2689](https://github.com/diegomura/react-pdf/pull/2689)

🙏 Thank you for your PRs @joelybahh, both issues cleared up for me with the latest version! And big thanks to all who maintain these packages!
